geth/node/manager.go

```diff
@@ -518,3 +519,64 @@ func (m *NodeManager) isNodeAvailable() error {
 
 	return nil
 }
+
+// tickerResolution is the delta to check blockchain sync progress.
+const tickerResolution = time.Second
+
+// Sync waits until blockchain synchronization and returns.
+// Timeout must be at least two seconds and zero timeout
+// means waiting infinitely.
+func (m *NodeManager) Sync(timeout time.Duration) error {
+	// We need to have a larger timeout than ticker delta here
+	// unless we use zero which means infinite timeout.
+	if timeout < tickerResolution*2 && timeout != 0 {
+		return errors.New("Sync timeout can only be zero (infinite) or at least two seconds")
```

Comment:
Usually, errors start with a lower case letter for easier joining (`log.Fatalf("sync error: %v", err)`) and as we talked in a different thread, maybe we can rephrase it to: `invalid timeout that can be zero (meaning no timeout) or greater than two seconds`.

However, I would also get rid of this `at least two seconds`. If, for instance, the chain is already synced, I should not wait at all. It can be easily achieved by running a check before `for` loop.